cf-agent/verify_files.c

```diff
@@ -755,19 +793,36 @@ PromiseResult ScheduleEditOperation(EvalContext *ctx, char *filename, Attributes
         }
     }
 
-
-    if (a.edit_template)
+    if (strcmp("cfengine", a.template_method) == 0)
```

Comment:
Check if `a.template_method == NULL` before `strcmp`.

Reply:
Disregard above comment, since you have added a check in `GetFilesAttributes()`.